acme_diags/driver/qbo_driver.py

```diff
@@ -0,0 +1,222 @@
+    # Average over latitude (i.e., average for entire data_region)
+    data_lon_lat_average = cdutil.averager(data_lon_average, axis='y')
+    # Get data by vertical level
+    level_data = data_lon_lat_average.getAxis(1)
+    return data_lon_lat_average, level_data
+
+
+def deseason(xraw):
+    # Calculates the deseasonalized data
+    months_per_year = 12
+    # Create array to hold climatological values and deseasonalized data
+    # Create months_per_year x 1 array of zeros
+    xclim = np.zeros((months_per_year, 1))
+    # Create array with same shape as xraw
+    x_deseasoned = np.zeros(xraw.shape)
+    # Iterate through all 12 months.
+    for month in np.arange(months_per_year):
+        # `xraw[month::12]` will return the data for this month every year (12 months)
+        # (i.e., from month until the end of xraw, get every 12th month)
+        # Get the mean of this month, using data from every year, ignoring NaNs
+        xclim[month] = np.nanmean(xraw[month::months_per_year])
+    num_years = int(np.floor(len(x_deseasoned) / months_per_year))
+    # Iterate through all years in x_deseasoned (same number as in xraw)
+    for year in np.arange(num_years):
+        year_index = year * months_per_year
+        # Iterate through all months of the year
+        for month in np.arange(months_per_year):
+            month_index = year_index + month
+            # Subtract the month's mean over num_years from xraw's data for this month in this year
+            # i.e., get the difference between this month's value and it's "usual" value
+            x_deseasoned[month_index] = xraw[month_index] - xclim[month]
+    return x_deseasoned
+
+
+def get_20to40month_fft_amplitude(qboN, levelN):
+    # Calculates the amplitude of wind variations in the 20 - 40 month period
+    psd_sumN = np.zeros(levelN.shape)
+    amplitudeN = np.zeros(levelN.shape)
+
+    for ilev in np.arange(len(levelN)):
+        # `qboN[:, ilev]` returns the entire 0th dimension for ilev in the 1st dimension of the array.
+        y_input = deseason(np.squeeze(qboN[:, ilev]))
+        y = scipy.fftpack.fft(y_input)
+        n = len(y)
+        frequency = np.arange(n / 2) / n
+        period = 1 / frequency
+        values = y[0:int(np.floor(n / 2))]
+        fyy = values * np.conj(values)
+        # Choose the range 20 - 40 months that captures most QBOs (in nature)
+        psd_sumN[ilev] = 2 * np.nansum(fyy[(period <= 40) & (period >= 20)])
+        amplitudeN[ilev] = np.sqrt(2 * psd_sumN[ilev]) * (frequency[1] - frequency[0])
+    return psd_sumN, amplitudeN
+
+
+def process_u_for_power_spectral_density(data_region):
+    # Average over vertical levels and horizontal area
+    levBot = 22
+    levTop = 18
+    # Average over lat and lon
+    data_lat_lon_average = cdutil.averager(data_region, axis='xy')
+    # Average over vertical
+    x0 = np.nanmean(np.array(data_lat_lon_average(level=(levTop, levBot))), axis=1)
```

Comment:
As @crterai commented, if no levels that exist between 18hPa and 22hPa, this algorithm would fail. Can we make it a try-except clause, and raise error message when "No levels that exist between 18hPa and 22hPa"